src/tqec/compile/compile.py

```diff
@@ -43,8 +43,10 @@ def __post_init__(self) -> None:
                 "The compiled graph should have at least one time slice "
                 "but got an empty layout_slices."
             )
+        # Use warning instead of exception because a qec circuit without observable
+        # may still be useful, e.g. do statistics on the detection events.
         if len(self.observables) == 0:
-            raise TQECException("The compiled graph includes no observable.")
+            raise TQECWarning("The compiled graph includes no observable.")
```

Comment:
Warnings are supposed to be used with `warnings.warn`. I am not sure this line has the meaning you intended, as I think it will behave just like a regular exception.